.github/workflows/build.yml

```diff
@@ -26,10 +26,11 @@ permissions:
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
-  cancel-in-progress: ${{ github.ref_name != 'main' }}
+  cancel-in-progress: true
```

Comment:
Would this rule be reverted later, or it's not actually necessary to begin with?

Reply:
it's no longer necessary, semantic release will break anyways if a new commit is pushed to main, so we should cancel early. there is a small chance to cancel the job while a release is published, but i havn't seen it on our other repos.